src/coreclr/tools/Common/Internal/Text/Utf8StringBuilder.cs

```diff
@@ -57,17 +56,11 @@ public Utf8StringBuilder Append(char value)
 
         public Utf8StringBuilder Append(string value)
```

Comment:
Eventually, I managed to figure out a way to access JIT's machinery for writing out pre-converted `string` literals to UTF-8 without `UnsafeAccessor`: https://github.com/U8String/U8String/blob/main/Sources/U8String/Shared/U8Interpolation.cs#L250-L270

If you annotate this method with `AggressiveInlining` and copy the implementation for `U8String`'s interpolated handler guts, this should completely optimize away the conversion (at the cost of not knowing `length`, but standard 2x is good enough).

Reply:
Now got on my computer so I can properly look at this.

It's really good idea, and I will definitely keep the tricks that you use in your library in mind in my future projects 😄 

Hopefully https://github.com/dotnet/csharplang/issues/7072 will make our hacks obsolete _someday_.